src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -174,12 +153,66 @@ private boolean hasMatchingRepository(RefChangedDetails refChangedDetails,
                 }).orElse(false);
     }
 
+    private boolean isEligibleRefs(RefsChangedWebhookEvent event) {
+        if (eligibleRefs(event).isEmpty()) {
+            LOGGER.fine("Skipping processing of refs changed event because no refs have been added or updated");
+            return false;
+        }
+        return true;
+    }
+
+    private void triggerJob(RefsChangedWebhookEvent event,
+                            RefChangedDetails refChangedDetails) {
+        try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+            BitbucketWebhookTriggerRequest.Builder requestBuilder = BitbucketWebhookTriggerRequest.builder();
+            event.getActor().ifPresent(requestBuilder::actor);
+
+            Jenkins.get().getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)
+                    .stream()
+                    .map(BitbucketWebhookConsumer::toTriggerDetails)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .filter(triggerDetails -> hasMatchingRepository(refChangedDetails, triggerDetails.getJob()))
+                    .peek(triggerDetails -> LOGGER.fine("Triggering " + triggerDetails.getJob().getFullDisplayName()))
+                    .forEach(triggerDetails -> triggerDetails.getTrigger().trigger(requestBuilder.build()));
+            //fire the head event to indicate to the SCMSources that changes have happened.
+            BitbucketSCMHeadEvent.fireNow(new BitbucketSCMHeadEvent(SCMEvent.Type.UPDATED, event, event.getRepository().getSlug()));
+        }
+    }
+
+    private static class BitbucketSCMHeadEvent extends SCMHeadEvent<RefsChangedWebhookEvent> {
```

Comment:
I assume this is just moved?

Reply:
Yes, nearly this whole file is just moved. The only change I made is to the `testScms` method. Sorry, should have clarified that in the message